manim/mobject/text/tex_mobject.py

```diff
@@ -381,6 +381,29 @@ def set_color_by_tex(self, tex, color, **kwargs):
             part.set_color(color)
         return self
 
+    def set_opacity_by_tex(self, tex, opacity=0.5, remaining_opacity=None, **kwargs):
+        """
+        Sets the opacity of the tex specified. If 'remaining_opacity' is specified,
+        then the remaining tex will be set to that opacity.
+
+        Parameters
+        ----------
+        tex : str
+            The tex to set the opacity of.
+        opacity : float, optional
+            Default 0.5. The opacity to set the tex to
+        remaining_opacity : float, optional
+            Default None. The opacity to set the remaining tex to.
+            If None, then the remaining tex will not be changed
+        """
+        part_index = self.index_of_part_by_tex(tex)
```

Comment:
With this approach, you will only be able to select the specified tex (sub)string a single time, further occurrences are not considered (as `index_of_part_by_tex` only returns the index of the first match).

You could change this to the previous implementation by doing something like this:
```
if remaining_opacity is not None:
    self.set_opacity(opacity=remaining_opacity)
for part in self.get_parts_by_tex(...):
    part.set_opacity(opacity)
```

What do you think?